src/components/Payroll/RunPayrollFlow/RunPayrollFlow.stories.tsx

```diff
@@ -0,0 +1,406 @@
+    <PayrollConfiguration
+      employees={[{ employeeId: 'cdef' }]}
+      onBack={action('on_back')}
+      onCalculatePayroll={action('on_calculate')}
+      onEdit={action('on_edit')}
+    />
+  )
+}
+
+export const PayrollEditEmployee = ({ onDone }) => {
+  const { Button, Heading, Text } = useComponentContext()
+  const formHandlers = useForm()
+  return (
+    <Flex flexDirection="column" gap={20}>
+      <Heading as="h2">Edit Hannah Arendt's payroll</Heading>
+      <Heading as="h1">$1,173.08</Heading>
+      <Text>Gross pay</Text>
+      <Heading as="h3">Regular hours</Heading>
+      <FormProvider {...formHandlers}>
+        <Form>
+          <NumberInputField defaultValue={40} isRequired label="Hours" name="hours" />
+        </Form>
+      </FormProvider>
+
+      <Button onClick={onDone} title="Done">
+        Done
+      </Button>
+    </Flex>
+  )
+}
+
+export const PayrollOverview = ({ onEdit, onSubmit }) => {
+  const { Alert, Button, Heading, Text } = useComponentContext()
+
+  return (
+    <Flex flexDirection="column" alignItems="stretch">
+      <Flex justifyContent="space-between">
+        <Heading as="h1">Review payroll for Jul 5 - Jul 18, 2025</Heading>
+        <Flex justifyContent="flex-end">
+          <Button title="Edit" onClick={onEdit} variant="secondary">
+            Edit
+          </Button>
+          <Button title="Submit" onClick={onSubmit}>
+            Submit
+          </Button>
+        </Flex>
+      </Flex>
+      <Alert label="Your progress has been saved" status="success"></Alert>
+      <Alert
+        label="To pay your employees with direct deposit on Fri, Jul 25, you'll need to run payroll by 7:00 PM EDT on Wed, Jul 23"
+        status="warning"
+      >
+        {"If you miss this deadline, your employees' direct deposit will be delayed."}
+      </Alert>
+      <Heading as="h3">Payroll Summary</Heading>
+      <DataView
+        label="Summary"
+        columns={[
+          {
+            title: 'Total payroll',
+            render: () => <Text>$32,161.22</Text>,
+          },
+          {
+            title: 'Debit amount',
+            render: () => <Text>$28,896.27</Text>,
+          },
+        ]}
+        data={[{}]}
+      />
+      <DataView
+        label="Configuration"
+        columns={[
+          {
+            title: 'Employees',
+            render: () => <Text>John Smith</Text>,
+          },
+          {
+            title: 'Gross Pay',
+            render: () => <Text>$2,345.16</Text>,
+          },
+          {
+            title: 'Reimbursements',
+            render: () => <Text>$0.00</Text>,
+          },
+        ]}
+        data={[{}]}
+      />
+    </Flex>
+  )
+}
+
+// **** BLOCKS ****
+// **** BLOCKS ****
+// **** BLOCKS ****
+
+// TODO: Replace this hook with call to Speakeasy instead
+const useEditEmployeeApi = ({ employeeId }) => {
+  const mutate = async () => {}
+  return { mutate }
+}
+const PayrollEditEmployeeBlock = ({ employeeId, onEvent, onSaved }) => {
```

Comment:
Do we actually need `onSaved` exposed here? or could we just use the `onEvent` prop?

Reply:
yeah i think it's an interesting discussion. I'd maybe favor just uniformly using onEvent, but i'd be open to arguments to have better named callbacks. In the case of callbacks that also would provide better type safety vs. onEvent which just kinda gets any data passed along

Reply:
I'm OK with using onEvent as well here, but suspect it would lead to our next discussion being "why do we use onEvent everywhere instead of discrete event handlers" and I'm maybe jumping ahead a step here